gio/src/action_entry.rs

```diff
@@ -87,6 +88,25 @@ where
         self
     }
 
+    pub fn activate_async<Fut, F>(mut self, callback: F) -> Self
+    where
+        F: Fn(&O, &SimpleAction, Option<&Variant>) -> Fut + 'static + Clone,
+        Fut: Future<Output = ()>,
+    {
+        let future_cb = move |map: &O, action: &SimpleAction, variant: Option<&Variant>| {
+            let ctx = glib::MainContext::default();
```

Comment:
This is potentially problematic. Do we always want to run things on the global default main context here, or could this also be on the thread-default one? See also https://github.com/gtk-rs/gtk-rs-core/pull/902